core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -531,7 +531,7 @@ else if(segment.startsWith("{") && segment.endsWith("}")) {
 					log.debug("calculating etags over processed result");
 					String cleanPattern = listener.getCleanPattern();
 					if(!Message.isEmpty(result) && method == HttpMethod.GET && cleanPattern != null) { //If the data has changed, generate a new eTag
-						String eTag = ApiCacheManager.buildEtag(cleanPattern, result.asObject().hashCode()); //The eTag has nothing to do with the content and can be a random string.
+						String eTag = ApiCacheManager.buildEtag(cleanPattern, result.hashCode()); //The eTag has nothing to do with the content and can be a random string.
```

Comment:
De etag moet wel veranderen als de content verandert. Deze implementatie lijkt me van het type 'gaat meestal wel goed'. Vinden we dat goed genoeg?

Reply:
message neemt de hashcode van het object over :)